### ISSUE TITLE
Don’t always add "defaultIgnorePatterns"

### ISSUE THREAD

**Issue by rluba** [BrowserSync/browser-sync#1543]
### Issue details

`browser-sync` introduced a breaking change in `v2.22.0`: [It added a list of `defaultIgnorePatterns` that are not only default, but _always ignored_.](https://github.com/BrowserSync/browser-sync/blob/f72a6ad619d3c9abe27905e15ef5b14b480c8b46/lib/cli/transforms/addDefaultIgnorePatterns.ts)

We are often developing with symlinked packages (via `npm link`/`bower link`). We intentionally watch certain files in `node_modules` and `bower_components` to ease working across package boundaries. This broke in `v2.22.0` since these files are now always ignored. There’s no way to disable these "default" ignore patterns—they are unconditionally merged with any ignore patterns defined by the user.

### Please specify which version of Browsersync, node and npm you're running

- Browsersync [v2.23.7]
- Node        [v8.9.1]
- Npm         [v5.8.0]

### Affected platforms

- [x] linux
- [x] windows
- [x] OS X
- [ ] freebsd
- [ ] solaris
- [ ] other _(please specify which)_

(But it will probably affect the others as well…)

### Browsersync use-case

- [x] API
- [x] Gulp
- [ ] Grunt
- [ ] CLI

(But it will probably affect the others as well…)

**Comment by shakyShane** [comment]
@rluba Hi!

This was not technically a breaking change, since it was introduced only when the new 'watch' option is used - although I can see why it appears to be.

The simplest solution for you, is to remove `watch: true` from your configuration, and instead specifiy the files you want to watch (this is how Browsersync always worked before the addition of `watch`)

```js
// before
browserSync.init({
  server: "./app",
  watch: true
}}

// after
browserSync.init({
  server: "./app",
  files: [
      "./app",
      "node_modules/some/dir",
      "bower_components" 
  ]
}}
```

or to just watch everything 

```
browserSync.init({
  server: "./app",
  files: ["."]
}}
```

**Comment by shakyShane** [comment]
https://github.com/BrowserSync/browser-sync/blob/f72a6ad619d3c9abe27905e15ef5b14b480c8b46/lib/cli/transforms/addDefaultIgnorePatterns.ts#L13-L15

**Comment by rluba** [comment]
Hi @shakyShane,

that’s what I initially expected, but we didn’t provide the `watch` option initially (since it’s new). We just noticed our watches breaking when upgrading past `v2.21.0`. I had a look at the doc, saw the new `watch` option (and it’s default, `false`) and tried setting it – it makes no difference.

Our usage looks like this:
``` JavaScript
            const server = browserSync.create('App');
            server.init(files, {
                startPath: '/index.html',
                server: {
                    baseDir: baseDirs,
                    middleware: proxy(config) // a partial HTTP proxy, doesn’t affect this issue
                },
                open: false,
                ghostMode: false
            }, callback);
```

with

``` JavaScript
baseDirs = [ 'build/.tmp', 'app' ];
files = [
    'app/src/**/*.html',
    <some files benath node_modules or bower_components>
    'build/.tmp/**/*' 
];
```

Touching any files in `app` or `build/.tmp` triggers a refresh, but nothing happens when we touch any of the files in `bower_components`.

The exact same configuration works when downgrading to `v2.21.0` and is broken since `v2.22.0`, whether you add `watch: true` or not.

**Comment by shakyShane** [comment]
@rluba thanks for the detail 

In that case it does seem like a bug, I'll investigate asap!

:)

**Comment by shakyShane** [comment]
Now that 2.24.0 is out, I'll be getting to this next :)

**Comment by shakyShane** [comment]
@rluba I've found the bug, I'll release a fix this evening :)

**Comment by shakyShane** [comment]
@rluba I just published a release candidate - could you try it our on your projects and let me know if it works :)

I was able to reproduce in both unit & integration tests, so I'm confident - but since you raised the bug I'd appreciate the verification

Please try browser-sync@2.24.3-rc1

```
npm install browser-sync@rc
```

or 

```
yarn add browser-sync@rc
```

**Comment by shakyShane** [comment]
Fixed in browser-sync@2.24.3

**Comment by rluba** [comment]
It works again. Thank you! ✌️
